docs/multicluster/setup-multi-primary.sh

```diff
@@ -0,0 +1,196 @@
+      --from-file=certs/east/ca-cert.pem \
+      --from-file=certs/east/ca-key.pem \
+      --from-file=certs/east/root-cert.pem \
+      --from-file=certs/east/cert-chain.pem
+
+kubectl get ns istio-system --context "${CTX_CLUSTER2}" || kubectl create namespace istio-system --context "${CTX_CLUSTER2}"
+kubectl --context "${CTX_CLUSTER2}" label namespace istio-system topology.istio.io/network=network2
+kubectl get secret --context "${CTX_CLUSTER2}" cacerts || kubectl create secret generic cacerts -n istio-system --context "${CTX_CLUSTER2}" \
+      --from-file=certs/west/ca-cert.pem \
+      --from-file=certs/west/ca-key.pem \
+      --from-file=certs/west/root-cert.pem \
+      --from-file=certs/west/cert-chain.pem
+
+
+# 4. Create Sail CR on east
+
+kubectl apply --context "${CTX_CLUSTER1}" -f - <<EOF
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.22.0
+  namespace: istio-system
+  values:
+    pilot:
+      resources:
+        requests:
+          cpu: 100m
+          memory: 1024Mi
+    global:
+      meshID: mesh1
+      multiCluster:
+        clusterName: east
+      network: network1
+EOF
+
+kubectl wait --context "${CTX_CLUSTER1}" --for=jsonpath='{.status.revisions.ready}'=1 istios/default --timeout=3m
+
+# 5. Create east-west gateway on east
+
+"${ISTIOCTL_DIR}"/samples/multicluster/gen-eastwest-gateway.sh \
+    --mesh mesh1 --cluster east --network network1 | \
+    "${ISTIOCTL_DIR}/bin/istioctl" --context="${CTX_CLUSTER1}" install -y -f -
+
+# 6. Expose services on east
+
+kubectl --context "${CTX_CLUSTER1}" apply -n istio-system -f \
+    "${ISTIOCTL_DIR}"/samples/multicluster/expose-services.yaml
+
+# 7. Create Sail CR on west
+
+kubectl apply --context "${CTX_CLUSTER2}" -f - <<EOF
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.22.0
+  namespace: istio-system
+  values:
+    pilot:
+      resources:
+        requests:
+          cpu: 100m
+          memory: 1024Mi
+    global:
+      meshID: mesh1
+      multiCluster:
+        clusterName: west
+      network: network2
+EOF
+
+kubectl wait --context "${CTX_CLUSTER2}" --for=jsonpath='{.status.revisions.ready}'=1 istios/default --timeout=3m
+
+# 8. Create east-west gateway on west
+
+"${ISTIOCTL_DIR}"/samples/multicluster/gen-eastwest-gateway.sh \
+    --mesh mesh1 --cluster west --network network2 | \
+    "${ISTIOCTL_DIR}"/bin/istioctl --context "${CTX_CLUSTER2}" install -y -f -
+
+# 9. Expose services on west
+
+kubectl --context "${CTX_CLUSTER2}" apply -n istio-system -f \
+    "${ISTIOCTL_DIR}"/samples/multicluster/expose-services.yaml
+
+# 10. Install a remote secret in west that provides access to east’s API server.
+
+WEST_CONTAINER_IP=$(kubectl get nodes east-control-plane --context "${CTX_CLUSTER1}" -o jsonpath='{.status.addresses[?(@.type == "InternalIP")].address}')
```

Comment:
Please see my earlier comment about using the shorter version of the `istioctl create-remote-secret ... ` command.

Reply:
The `--server` argument is required when running on kind.